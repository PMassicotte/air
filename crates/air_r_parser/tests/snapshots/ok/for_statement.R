for (i in 1) i
for (
  i # comment
  in # comment2
  1 # comment3
) i
