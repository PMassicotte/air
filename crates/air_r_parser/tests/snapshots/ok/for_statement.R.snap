---
source: crates/air_r_parser/tests/spec_test.rs
expression: snapshot
---
## Input

```R
for (i in 1) i
for (
  i # comment
  in # comment2
  1 # comment3
) i

```


## AST

```
RRoot {
    bom_token: missing (optional),
    expressions: RExpressionList [
        RForStatement {
            for_token: FOR_KW@0..3 "for" [] [],
            l_paren_token: L_PAREN@3..5 "(" [Whitespace(" ")] [],
            variable: RIdentifier {
                name_token: IDENT@5..6 "i" [] [],
            },
            in_token: IN_KW@6..9 "in" [Whitespace(" ")] [],
            sequence: RDoubleValue {
                value_token: R_DOUBLE_LITERAL@9..11 "1" [Whitespace(" ")] [],
            },
            r_paren_token: R_PAREN@11..12 ")" [] [],
            body: RIdentifier {
                name_token: IDENT@12..14 "i" [Whitespace(" ")] [],
            },
        },
        RForStatement {
            for_token: FOR_KW@14..18 "for" [Newline("\n")] [],
            l_paren_token: L_PAREN@18..20 "(" [Whitespace(" ")] [],
            variable: RIdentifier {
                name_token: IDENT@20..34 "i" [Newline("\n"), Whitespace("  ")] [Whitespace(" "), Comments("# comment")],
            },
            in_token: IN_KW@34..50 "in" [Newline("\n"), Whitespace("  ")] [Whitespace(" "), Comments("# comment2")],
            sequence: RDoubleValue {
                value_token: R_DOUBLE_LITERAL@50..65 "1" [Newline("\n"), Whitespace("  ")] [Whitespace(" "), Comments("# comment3")],
            },
            r_paren_token: R_PAREN@65..67 ")" [Newline("\n")] [],
            body: RIdentifier {
                name_token: IDENT@67..69 "i" [Whitespace(" ")] [],
            },
        },
    ],
    eof_token: EOF@69..70 "" [Newline("\n")] [],
}
```

## CST

```
0: R_ROOT@0..70
  0: (empty)
  1: R_EXPRESSION_LIST@0..69
    0: R_FOR_STATEMENT@0..14
      0: FOR_KW@0..3 "for" [] []
      1: L_PAREN@3..5 "(" [Whitespace(" ")] []
      2: R_IDENTIFIER@5..6
        0: IDENT@5..6 "i" [] []
      3: IN_KW@6..9 "in" [Whitespace(" ")] []
      4: R_DOUBLE_VALUE@9..11
        0: R_DOUBLE_LITERAL@9..11 "1" [Whitespace(" ")] []
      5: R_PAREN@11..12 ")" [] []
      6: R_IDENTIFIER@12..14
        0: IDENT@12..14 "i" [Whitespace(" ")] []
    1: R_FOR_STATEMENT@14..69
      0: FOR_KW@14..18 "for" [Newline("\n")] []
      1: L_PAREN@18..20 "(" [Whitespace(" ")] []
      2: R_IDENTIFIER@20..34
        0: IDENT@20..34 "i" [Newline("\n"), Whitespace("  ")] [Whitespace(" "), Comments("# comment")]
      3: IN_KW@34..50 "in" [Newline("\n"), Whitespace("  ")] [Whitespace(" "), Comments("# comment2")]
      4: R_DOUBLE_VALUE@50..65
        0: R_DOUBLE_LITERAL@50..65 "1" [Newline("\n"), Whitespace("  ")] [Whitespace(" "), Comments("# comment3")]
      5: R_PAREN@65..67 ")" [Newline("\n")] []
      6: R_IDENTIFIER@67..69
        0: IDENT@67..69 "i" [Whitespace(" ")] []
  2: EOF@69..70 "" [Newline("\n")] []

```
